# ITMGT 25.03 EE
# Assignment 3A

This is a data-cleaning assignment and is the first of two parts. The second part (3B) will be given Monday or Tuesday next week. This notebook is uploaded to our Github repository along with a csv of fake, dirty data called **recweekdata.csv**.


The output of this assignment is to convert recweekdata.csv into processable data for statistics. Submit the following files through GitHub:
1. A cleaned recweekdata.csv called cleanrecweekdata.csv **(5 points)**
2. This notebook, containing the code (properly commented) that you used to clean the data **(5 points)**
3. An html file of summary statistics from each column called **stats.html**. The requirements for the summary statistics are described below. **(10 points)**

This entire assignment (3A) is worth 20 points.

Upload your output files to your Github repositories. Make sure to send an email to **joben.ilagan@gmail.com** once you're done.

This assignment is due on **Tuesday, July 9, 2019** at 12:00NN.

In [5]:
## Code goes here
import csv

data_dict = dict()
with open ("recweekdata.csv") as csv_file:
    dirty_recweek = csv.reader(csv_file, delimiter = ",")
    line_count = 0
    for row in dirty_recweek:
        if line_count == 0:
            pass
        else:
            data_dict[(row[0])[4:]] = {"name": row[1], "bmonth": row[2], "bday": row[3], "byear": row[4], "course": row[5], "number": row[6]}
        line_count += 1
    print(recweek_dict)

{'1': {'name': 'lastname1', 'bmonth': '11', 'bday': '26', 'byear': '1998', 'course': 'BS LM', 'number': '+63 918 302 6210'}, '2': {'name': 'lastname2', 'bmonth': 'May', 'bday': '26', 'byear': '2001', 'course': 'Management', 'number': '927 212 6276'}, '3': {'name': 'lastname3', 'bmonth': '5', 'bday': '16', 'byear': '1997', 'course': 'BS CTM', 'number': '+63 918 253 7017'}, '4': {'name': 'lastname4', 'bmonth': 'Dec', 'bday': '9', 'byear': '97', 'course': 'BS ITE', 'number': '918 714 4360'}, '5': {'name': 'lastname5', 'bmonth': 'Sep', 'bday': '5', 'byear': '97', 'course': 'BS CTM', 'number': '+63 927 462 1674'}, '6': {'name': 'lastname6', 'bmonth': 'March', 'bday': '2', 'byear': '1999', 'course': 'Management', 'number': '+63 918 629 5268'}, '7': {'name': 'lastname7', 'bmonth': '11', 'bday': '10', 'byear': '98', 'course': 'BS LM', 'number': '+63 917 369 7388'}, '8': {'name': 'lastname8', 'bmonth': 'Aug', 'bday': '28', 'byear': '99', 'course': 'Management Engineering', 'number': '917 240 60

In [31]:
import re
month_dict = {1:"January",2:"February",3:"March",4:"April",5:"May",6:"June",7:"July",8:"August",9:"September",10:"October",11:"November",12:"December"}

for row in range(1, len(data_dict)+1):
    if len(str(data_dict[str(row)]["bmonth"]))==1:
        for i in month_dict:
            recweek_dict[str(row)]["bmonth"] = re.sub(r'[i]',r''.join(month_dict[i]),recweek_dict[str(row)]["bmonth"])
    elif len(str(data_dict[str(row)]["bmonth"]))==2:
        for i in range(10,13):
            recweek_dict[str(row)]["bmonth"] = re.sub(r'[i]',r''.join(month_dict[i]),recweek_dict[str(row)]["bmonth"])
    print(recweek_dict[str(row)]["bmonth"])

11
May
January
Dec
Sep
March
11
Aug
July
January
December
January
January
January
January
January
January
Apr
12
January
January
January
January
Oct
January
12
March
11
10
May
January
Jul
12
Oct
January
December
January
January
May
January
Nov
September
Jan
February
January
Dec
August
Apr
10
March
January
January
Dec
January
May
January
May
April
April
August
January
Dec
Feb
November
Mar
September
Dec
November
November
May
January
December
Jun
Oct
Sep
January
May
January
December
January
Feb
11
Jun
February
Jan
Nov
August
January
January
Mar
Aug
Dec
Jun
November
January
Mar
Jan
Jan
May
January
April
August
Dec
July
Mar
January
April
May
January
January
12
May
May
August
March
Jul
January
Feb
Dec
January
January
March
January
February
November
June
Mar
June
10
Jul
Feb
January
Apr
May
10
10
June
February
April
Sep
Jun
12
Nov
Oct
January
Sep
December
November
January
Aug
May
January
November
Aug
12
January
August
January
Oct
12
Jan
January
April
May
Aug
Feb
January
January
January
Jun
Aug

January
Jul
Jan
January
January
January
Jun
April
September
January
January
Nov
April
January
Aug
Jan
Jun
January
Oct
Aug
Feb
January
May
Sep
January
May
Oct
Jan
Oct
January
Sep
January
February
May
September
Apr
Apr
January
January
May
Oct
January
August
March
May
January
June
January
January
November
Nov
Dec
January
11
Dec
Feb
January
12
Apr
April
May
Feb
January
January
January
10
April
January
Aug
Aug
January
January
January
January
January
November
Jul
Aug
May
January
May
January
June
July
January
Apr
Nov
February
Apr
January
January
December
January
May
September
January
November
Aug
11
January
August
Jul
Jun
January
Jan
Jun
January
12
March
Dec
April
Jun
Nov
January
June
April
January
September
January
January
Dec
January
Jun
Jun
May
July
February
January
Nov
January
January
August
Jul
Jul
Aug
January
Apr
January
May
10
Aug
11
August
April
January
11
Jul
January
Jan
10
November
August
Apr
Nov
Jan
January
11
January
Mar
11
January
June
Jan
December
September
January
Nov
Mar
Dec
1

# Required Summary Statistics
1. How many people were born in each month?
2. How many people were born on each day?
3. How many people were born in each year?
4. How many people come from each course?
5. How many people have numbers from Globe (0917 or 0927) and Smart (0918)? *Note: it's up to you how to format this output*

Answer these questions in an html file called stats.html. The format of the html is as follows (in template form, so you can use this as format string in your program, though feel free to modify with variable names if you find it easier to do so):


```
<html>
    <body>
         <h2>Summary Statistics</h2>
         <table>
             <tr><th>Stat</th><th>Value</th></tr>
             <tr><td>People born in each month</td><td>{}</td></tr>
             <tr><td>People born on each day</td><td>{}</td></tr>
             <tr><td>People born in each year</td><td>{}</td></tr>
             <tr><td>People coming from each course</td><td>{}</td></tr>
             <tr><td>People having numbers from Globe and Smart</td><td>{}</td></tr>
         </table>
    </body>
</html>
```